traversal/scanner/GraphIterator.java

```diff
@@ -226,27 +225,32 @@ private void traverse(ProcedureVertex<?, ?> procedureVertex) {
     }
 
     private void success(VertexTraverser vertexTraverser) {
-        for (ProcedureVertex<?, ?> vertex : vertexTraverser.procedureVertex.dependents()) {
-            toTraverse.add(vertex);
-            vertexTraversers.get(vertex).clear();
-        }
-        for (ProcedureVertex<?, ?> vertex : vertexTraverser.implicitDependents) {
-            toTraverse.add(vertex);
-            vertexTraversers.get(vertex).clear();
-        }
```

Comment:
essentially, as explained in the PR description, we cannot skip forward just along the procedure edges because that can lead to inconsistent states. We must always proceed to the next vertex in the plan.
However, we can do some short-circuit backtracking when failing to find a vertex